### PR TITLE
zcbor 0.8.1

### PR DESCRIPTION
**Pull request by oyvindronningstad**

### CODE REVIEW

File: scripts/requirements-build.txt [hunk 1]
```diff
@@ -5,4 +5,4 @@ cryptography>=36.0.0
 imagesize>=1.2.0
 intelhex
 pylint
-zcbor
+zcbor>=0.8.0
```
Comment (SeppoTakalo) [inline scripts/requirements-build.txt:8]: Don't we limit this?
If 1.0.0 comes out and it changes API, it might break.

So should this be
```suggestion
zcbor>=0.8.1, == 0.8.*
```

And when 1.0 finally comes out, just (compatible release)
```
zbor ~= 1.0.0
```

**Review by SeppoTakalo — APPROVED** [review]

**Review by plskeggs — APPROVED** [review]

File: subsys/dfu/fmfu_fdev/cddl/CMakeLists.txt [hunk 2]
```diff
@@ -59,14 +59,16 @@ else()
    COMMAND ${CLANG_FORMAT} -i "${decode_c};${decode_h};${types_h}")
 endif()
 
+string(TIMESTAMP NOW_YEAR "%Y")
+
```
Comment (hakonfam) [inline subsys/dfu/fmfu_fdev/cddl/CMakeLists.txt:63]: Is this needed?

File: subsys/dfu/dfu_target/src/dfu_target_smp.c [hunk 3]
```diff
@@ -314,7 +314,7 @@ static int dfu_target_smp_recovery_enable(void)
 {
 	int rc;
 
-	rc = os_mgmt_client_echo(&os_gr_client, dfu_smp_echo_str);
+	rc = os_mgmt_client_echo(&os_gr_client, dfu_smp_echo_str, sizeof(dfu_smp_echo_str));
```
Comment (hakonfam) [inline subsys/dfu/dfu_target/src/dfu_target_smp.c:317]: How is this related to zcbor? (mostly curious)

File: subsys/dfu/fmfu_fdev/Kconfig [hunk 4]
```diff
@@ -15,6 +15,7 @@ if (FMFU_FDEV)
 
 config FMFU_CDDL_DECODER_GENERATE
 	bool
+	default y
```
Comment (hakonfam) [inline subsys/dfu/fmfu_fdev/Kconfig:18]: Why do we introduce this change?

**Review by juhaylinen — APPROVED** [review]

File: scripts/requirements-build.txt (hunk 1 above)
Comment (SeppoTakalo) [reply scripts/requirements-build.txt]: I don't think `~=` is safe with `0.x.x` because it is `x.y.z ~= x.*.*` so this would be same as `0.*.*`

It depends on how Zcbor team keeps the API compatible. Hopefully 0.10, if ever published, is going to be compatible.

Comment (oyvindronningstad) [reply scripts/requirements-build.txt]: I think it only means 0.8.x: https://peps.python.org/pep-0440/#compatible-release

**Comment by shanthanordic** [comment]
@oyvindronningstad  is this must needed for NCS 2.6.0 ?

File: scripts/requirements-build.txt (hunk 1 above)
Comment (SeppoTakalo) [reply scripts/requirements-build.txt]: Btw. I don't mean that this has to be changed now. Many other rules in our `requirements.txt` are equally problematic as they use `>=`
I just wanted to point out that this might cause issues in future.

**Comment by oyvindronningstad** [comment]
> is this must needed for NCS 2.6.0 ?

@shanthanordic I don't think so, but it is needed before the next upmerge of zephyr, since zcbor 0.8.1 is on zephyr main now.

**Review by hakonfam — APPROVED** [review]

**Review by shanthanordic — APPROVED** [review]
Thw downstream failures are env issues as agreed with @MarkusLassila  and @tony-le-24 . 
@carlescufi  or @rlubos  it is ok to merge this PR with RED CI. for IOT cloud and serial-lte domain

**Comment by MarkusLassila** [comment]
> The downstream failures are env issues as agreed with @MarkusLassila and @tony-le-24 . @carlescufi or @rlubos it is ok to merge this PR with RED CI. for IOT cloud and serial-lte domain

Actuall, SLM was broken with https://github.com/nrfconnect/sdk-nrf/pull/13871 and is currently broken in main-branch. It is not an environmental issue as such, but I don't think that merging this will have any effect on that, so +1 from SLM part.

**Review by MarkusLassila — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 5]
```diff
@@ -719,6 +725,13 @@ DFU libraries
 
     * The :kconfig:option:`CONFIG_DFU_TARGET_FULL_MODEM_USE_EXT_PARTITION` Kconfig option to be automatically enabled when ``nordic,pm-ext-flash`` is chosen in the devicetree.
       See :ref:`partition_manager` for details.
+    * Adapt to API changes in zcbor 0.8.x.
+
+* :ref:`lib_fmfu_fdev` library:
+
+  * Updated:
+
+    * Regenerate cbor code with zcbor 0.8.1 and adapt to API changes in zcbor 0.8.x.
```
Comment (umapraseeda) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:734]:
```suggestion
    * Regenerated cbor code with zcbor 0.8.1 and adapted to API changes in zcbor 0.8.x.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 6]
```diff
@@ -804,6 +817,10 @@ Modem libraries
 
   * Fixed a potential race condition, where two threads might corrupt a shared response buffer.
 
+* :ref:`lib_modem_attest_token` library:
+
+  * Adapt to API changes in zcbor 0.8.x.
```
Comment (umapraseeda) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:822]:
```suggestion
  * Updated to adapt to API changes in zcbor 0.8.x.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 7]
```diff
@@ -856,6 +873,8 @@ Libraries for networking
 
       This parameter determines whether CoAP CON or NON messages are used.
 
+    * Regenerate cbor code with zcbor 0.8.1 and adapt to API changes in zcbor 0.8.x.
```
Comment (umapraseeda) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:876]:
```suggestion
    * Regenerated cbor code with zcbor 0.8.1 and adapted to API changes in zcbor 0.8.x.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 8]
```diff
@@ -893,6 +912,7 @@ Libraries for networking
   * Updated the device mode callback to send an event when the provisioning state changes.
   * Fixed file descriptor handling by setting the :c:struct:`coap_client` structure's ``fd`` field to ``-1`` when closing the socket.
   * Added the :kconfig:option:`CONFIG_NRF_PROVISIONING_PRINT_ATTESTATION_TOKEN` option to enable printing the attestation token when the device is not yet claimed.
+  * Regenerate cbor code with zcbor 0.8.1 and adapt to API changes in zcbor 0.8.x.
```
Comment (umapraseeda) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:915]:
```suggestion
  * Regenerated cbor code with zcbor 0.8.1 and adapted to API changes in zcbor 0.8.x.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 9]
```diff
@@ -1030,7 +1050,17 @@ Additions specific to |NCS|
 zcbor
 =====
 
-|no_changes_yet_note|
+zcbor has been updated from 0.7.0 to 0.8.1.
+Full release notes can be found at:
+https://github.com/zephyrproject-rtos/zcbor/blob/0.8.0/RELEASE_NOTES.md and
+https://github.com/zephyrproject-rtos/zcbor/blob/0.8.1/RELEASE_NOTES.md
```
Comment (umapraseeda) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:1056]: Please add the entries : 
`zcbor 0.8.0 release notes`: https://github.com/zephyrproject-rtos/zcbor/blob/0.8.0/RELEASE_NOTES.md 
`zcbor 0.8.0 release notes`: https://github.com/zephyrproject-rtos/zcbor/blob/0.8.1/RELEASE_NOTES.md 
to doc/nrf/links.txt for the links to work.
```suggestion
For more information see the `zcbor 0.8.0 release notes`_ and the `zcbor 0.8.1 release notes`_.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 10]
```diff
@@ -1030,7 +1050,17 @@ Additions specific to |NCS|
...
+
+Highlights:
+
+* Add support for unordered maps
+* Performance improvements
+* Naming improvements for generated code
+* Bugfixes
```
Comment (umapraseeda) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:1063]:
```suggestion
* Added:

  * Support for unordered maps.
  * Performance improvements.
  * Naming improvements for generated code.
  * Bugfixes.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 11]
```diff
@@ -719,6 +725,13 @@ DFU libraries
 
     * The :kconfig:option:`CONFIG_DFU_TARGET_FULL_MODEM_USE_EXT_PARTITION` Kconfig option to be automatically enabled when ``nordic,pm-ext-flash`` is chosen in the devicetree.
       See :ref:`partition_manager` for details.
+    * Adapt to API changes in zcbor 0.8.x.
```
Comment (umapraseeda) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:728]:
```suggestion
    * Adapted to API changes in zcbor 0.8.x.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 8 above)
Comment (umapraseeda) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:915]:
```suggestion
  * Added the :kconfig:option:`CONFIG_NRF_PROVISIONING_PRINT_ATTESTATION_TOKEN` option to enable printing the attestation token when the device is not yet claimed.
  * Updated:
  
    * The device mode callback sends an event when the provisioning state changes.
    * Regenerated cbor code with zcbor 0.8.1 and adapted to API changes in zcbor 0.8.x.

  * Fixed file descriptor handling by setting the :c:struct:`coap_client` structure's ``fd`` field to ``-1`` when closing the socket.
```

Comment (oyvindronningstad) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:915]: Your last proposal changes text unrelated to my changes, can you confirm that I should change it here?

Comment (umapraseeda) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:915]: It is just grouping related points together. I can fix this later if needed.

Comment (oyvindronningstad) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: No worries, I added it, just wanted to confirm.

**Review by umapraseeda — APPROVED** [review]

**Review by KAGA164 — APPROVED** [review]
From nRF RPC perspective looks ok

**Comment by jfischer-no** [comment]
@oyvindronningstad Please update revisions/rebase

**Review by rlubos — APPROVED** [review]
